File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s :
            if i in ['(','{','[']:
                stack.append(i)
            else :
                if len(stack) == 0 : return False
                temp = stack.pop()
                if (i == ')' and temp != '(') or (i == ']' and temp != '[') or (i == '}' and temp != '{' ) : 
                    return False
                
        if len(stack) != 0 : return False
        return True

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<int> t;
        for(char c : s){
            if(c=='[' || c == '{' || c == '('){
                t.push(c);
            }else{
                if(t.empty()) return false;
                char a = t.top();
                t.pop();
                if((c==')' && a == '(')|| (c=='}' && a == '{') ||  (c==']' && a == '[')) continue;
                return false;
            }
        }   
        if(t.empty()) return true;
        return false;
    }
};